src/node_wasi.cc

```diff
@@ -0,0 +1,85 @@
+#include <cerrno>
+#ifdef _WIN32
+#include <io.h>
+#else
+#include <sys/types.h>
+#include <unistd.h>
+#endif
+
+#include "util-inl.h"
+#include "node_internals.h"
+
+#define NAPI_EXPERIMENTAL
+#include "node_api.h"
+
+namespace node {
+namespace wasi {
+
+napi_value Seek(napi_env env, napi_callback_info info) {
+  napi_status status;
+
+  napi_value args[3];
+  size_t argc = 3;
+  status = napi_get_cb_info(env, info, &argc, args, nullptr, nullptr);
+  if (status != napi_ok) {
+    return nullptr;
+  }
+
+  uint32_t fd;
+  status = napi_get_value_uint32(env, args[0], &fd);
+  if (status != napi_ok) {
+    return nullptr;
+  }
+
+  uint64_t offset;
+  status = napi_get_value_bigint_uint64(env, args[1], &offset, nullptr);
```

Comment:
> i attempted to add seeking but the native function keeps returning undefined on success instead of the new offset ;(.

I don't think the call ever actually succeeds. I'm seeing this call return `napi_invalid_arg` because the lossless argument can't be `nullptr`. Providing a `bool*` here makes the test pass for me.

Reply:
Here is the diff I'm using:

```diff
diff --git a/src/node_wasi.cc b/src/node_wasi.cc
index d40967fe9a..8b13585a99 100644
--- a/src/node_wasi.cc
+++ b/src/node_wasi.cc
@@ -32,7 +32,8 @@ napi_value Seek(napi_env env, napi_callback_info info) {
   }
 
   uint64_t offset;
-  status = napi_get_value_bigint_uint64(env, args[1], &offset, nullptr);
+  bool lossless;
+  status = napi_get_value_bigint_uint64(env, args[1], &offset, &lossless);
   if (status != napi_ok) {
     return nullptr;
   }
diff --git a/test/wasi/c/stat.c b/test/wasi/c/stat.c
index 4ad958dd00..7f7cab4f92 100644
--- a/test/wasi/c/stat.c
+++ b/test/wasi/c/stat.c
@@ -4,6 +4,7 @@
 #include <fcntl.h>
 #include <time.h>
 #include <unistd.h>
+#include <errno.h>
 
 #define BASE_DIR "/sandbox"
 #define OUTPUT_DIR BASE_DIR "/testdir"
@@ -15,17 +16,17 @@ int main(void)
     struct stat st;
     int         fd;
     int         ret;
-    // off_t       pos;
+    off_t       pos;
 
     (void) st;
     ret = mkdir(OUTPUT_DIR, 0755);
-    assert(ret == 0);
+    assert(ret == 0 || (ret == -1 && errno == EEXIST));
 
     fd = open(PATH, O_CREAT | O_WRONLY, 0666);
     assert(fd != -1);
 
-    // pos = lseek(fd, SIZE - 1, SEEK_SET);
-    // assert(pos == SIZE - 1);
+    pos = lseek(fd, SIZE - 1, SEEK_SET);
+    assert(pos == SIZE - 1);
 
     ret = (int) write(fd, "", 1);
     assert(ret == 1);

```